### ISSUE TITLE
cross-account rds-snapshot

### ISSUE THREAD

**Issue by pendyalal** [cloud-custodian/cloud-custodian#5657]
**Describe the bug**
Policy triggers on ModifyDBSnapshotAttribute event. Instead of checking a particular resource of the event, It describes all the RDS snapshots in the account
.
**To Reproduce**

```yaml
name: rds-shared-snapshot
resource: rds-snapshot
description: |
    Detects if an RDS Snapshot is shared with outside of the organization.
mode:
  type: cloudtrail
  role: Role
  events:
    - event: ModifyDBSnapshotAttribute
      source: rds.amazonaws.com
      ids: "requestParameters.dBSnapshotIdentifier"
filters:
 - type: cross-account
   whitelist_from:
           url: s3://accounts
           format: txt
```
**Expected behavior**
It should only check the modified RDS Snapshot. Not all snapshots in the account.

**Background (please complete the following information):**
 - OS: Amazon Linux 2
 - Python Version: python 3.7
 - Custodian Version: 0.8.46.1
 - Tool Version: [if applicable]
 - Cloud Provider:  aws
 - Policy: [please exclude any account/sensitive information]

```yaml
name: rds-shared-snapshot
resource: rds-snapshot
description: |
Detects if an RDS Snapshot is shared with outside of the organization.
mode:
  type: cloudtrail
  role: Role
  events:
    - event: ModifyDBSnapshotAttribute
      source: rds.amazonaws.com
      ids: "requestParameters.dBSnapshotIdentifier"
filters:
 - type: cross-account
   whitelist_from:
           url: s3://accounts
           format: txt
```

 - Traceback: [if applicable, please exclude sensitive/account information]
 - `custodian version --debug` output

**Additional context**

**Comment by JohnHillegass** [comment]
Thanks for logging @pendyalal. Could you also include the custodian run command you are using in this scenario?

**Comment by kapilt** [comment]
cli shouldn't be relevant

**Comment by csabella** [comment]
Would you be able to provide a debug log?

**Comment by pendyalal** [comment]
@csabella , Sorry I don't have the debug log. When I tested this couple months back, I had the problem.

**Comment by kapilt** [comment]
per private discussion, i think this was related to using resource group tagging which fetched tags for all extant snapshots, which is fixed since that api now supports fetching tags by arns and was merged in #6414
